src/snowflake/connector/arrow_iterator.pyx

```diff
@@ -88,6 +88,13 @@ cdef extern from "arrow/api.h" namespace "arrow" nogil:
     cdef cppclass CRecordBatchReader" arrow::RecordBatchReader":
         CStatus ReadNext(shared_ptr[CRecordBatch]* batch)
 
+    cdef cppclass CResult "arrow::Result"[T]:
+        CResult()
+        CResult(CStatus)
+        CResult(T)
+        c_bool ok()
+        CStatus status()
+        T operator*()
```

Comment:
Please look at line 81 and add the missing functions into the already existing definition.

Reply:
Out of curiosity: When aren't you using the offical `pxd` that is coming with `pyarrow`?

Reply:
I didn't write this code and since I took over it I have been just following the same patterns.
To be honest it works and so I didn't see too much value in getting myself to know all the ins and outs of the Arrow code (when people are asking for so many other things to be fixed).

But if you could give me some pointers I'd be happy to look into it some more!